### PR TITLE
[openshift] ignore owned resources

### PR DESCRIPTION
**Pull request by maorfr**
related to https://issues.redhat.com/browse/APPSRE-3356

With this PR, we can both act on owned resources (take over them) or ignore them (so they are not deleted)

### CODE REVIEW

**Comment by jfchevrette** [comment]
Simple and elegant!

I would make the argument default to `True` so the default is the same as what we will be running in prod (ignore owned, the least dangerous & most common case) and allow the rarer case of not wanting to ignore owner references via an argument (`--include-owned`?)

**Comment by maorfr** [comment]
@jfchevrette i've made this much simpler. care to take another look?

**Comment by jfchevrette** [comment]
@maorfr Beautiful! We can add the argument back if we ever need to act upon routes with owner refs

**Comment by maorfr** [comment]
With this, we can both act on them (take over them) and ignore them (so they are not deleted)!!

**Review by BumbleFeng — APPROVED** [review]

File: reconcile/utils/openshift_resource.py [hunk 1]
```diff
@@ -226,6 +226,9 @@ def has_qontract_annotations(self):
 
         return True
 
+    def has_owner_reference(self):
+        return len(self.body['metadata'].get('ownerReferences', [])) > 0
```
Comment (Piojo) [inline reconcile/utils/openshift_resource.py:230]: Simple nit:
```suggestion
        return bool(self.body['metadata'].get('ownerReferences', []))
```
which is closer to English' "the metadata has owner deferences".

**Review by Piojo — APPROVED** [review]
The nit is unimportant. Looks good, and thanks for the tests. :)

File: reconcile/utils/openshift_resource.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/openshift_resource.py:230]: makes sense. i really was going towards the bool direction. thanks!

**Pull request merged**